modules/01 JavaScript Programming/Assignments/27/answer27.js

```diff
@@ -0,0 +1,88 @@
+function Validator(value){
+
+this.isBoolean=function(){
+    return typeof value==='boolean';
+}
+this.isError=function(){
+    return value instanceof Error;
+}
+this.isNaNv =function(){
+    return isNaN(value);
+}
+this.isNull=function(){
+    return value=== null;
+}
+this.isNumberv = function(value){
+    return  typeof value == 'number';
+}
+this.isObject = function(value){
+    return (typeof value == 'object');
+}
+var isJsonPureObj= this.isPureJson = function(value)
+{
+    if (typeof value!=="string")
+        return false;
+        try{
+            JSON.parse(value);
+            return true;
+        }
+        catch (error){
+            return false;
+        }
+}
+
+this.isRegExp = function(value){return (value instanceof RegExp)}
+this.isChar = function(value){return ((value.length === 1) && ("abcdefghijklmnopqrstuvwxyz".includes(value)))}
```

Comment:
incorrect. you should include also capital letters in your check: "A", "B", "C" etc.